pkg/tfgen/installation_docs.go

```diff
@@ -245,3 +250,106 @@ func convertExample(g *Generator, code string, exampleNumber int) (string, error
 	exampleContent += chooserEnd
 	return exampleContent, nil
 }
+
+type SectionSkipper struct {
+	shouldSkipHeader shouldSkipHeaderFunc
+}
+
+var _ parser.ASTTransformer = SectionSkipper{}
+
+func (t SectionSkipper) Transform(node *ast.Document, reader text.Reader, pc parser.Context) {
+	source := reader.Source()
+	removingLevel := 0
+	currentChild := node.FirstChild()
+	// All headings are first children of the ast.Document node.
+	// We will walk over them and remove any that match the header content we do not want.
+	// Additionally, we will track the header level and remove any content that is nested under the removed header.
+	for currentChild != nil {
+		if removingLevel > 0 {
+			if child, ok := currentChild.(*ast.Heading); ok && child.Level <= removingLevel {
+				removingLevel = 0
+			} else {
+				nextNode := currentChild.NextSibling()
+				currentChild.Parent().RemoveChild(currentChild.Parent(), currentChild)
+				currentChild = nextNode
+				continue
+			}
+		}
+		// In case our level got reset, we check again
+		if removingLevel == 0 {
+			switch child := currentChild.(type) {
+			case *ast.Heading:
+				header := child.Text(source)
+				if t.shouldSkipHeader(string(header)) {
+					removingLevel = child.Level
+					nextNode := child.NextSibling()
+					child.Parent().RemoveChild(child.Parent(), child)
+					currentChild = nextNode
+					continue
+				}
+			}
+		}
+		// Move to next node in base case.
+		currentChild = currentChild.NextSibling()
+	}
+}
+
+type shouldSkipHeaderFunc = func(headerText string) bool
+
+// SkipSectionByHeaderContent takes a document's content byte array and a callback function that contains logic on
+// whether a given header should be skipped.
+// It uses "github.com/yuin/goldmark" parser in conjunction with "github.com/teekennedy/goldmark-markdown" renderer
+// to walk the markdown document tree and apply the SectionSkipper's transformer
+func SkipSectionByHeaderContent(content []byte, shouldSkipHeader shouldSkipHeaderFunc) ([]byte, error) {
```

Comment:
I don't see why an external user would want to call this function. Let's keep it private.

Reply:
This is public on purpose - when a provider would manually want to skip a more specific header, you'd be able to use this function to more easily create a docsEdit, similar to skipSectionHeadersEdit.